chalk-solve/src/solve/slg/aggregate.rs

```diff
@@ -229,9 +229,15 @@ impl<I: Interner> AntiUnifier<'_, '_, I> {
                 self.aggregate_application_tys(apply1, apply2)
             }
 
-            (TyData::Alias(alias1), TyData::Alias(alias2)) => {
-                self.aggregate_alias_tys(alias1, alias2)
-            }
+            (
+                TyData::Alias(AliasTy::Projection(proj1)),
+                TyData::Alias(AliasTy::Projection(proj2)),
+            ) => self.aggregate_projection_tys(proj1, proj2),
```

Comment:
I wonder if -- before this PR, at least -- this code was even live. I think we never *generate* "alias types" as part of an answer (only their associated placeholder), though we've considered dropping that invariant from time to time.